DatadogCore/Sources/Core/Storage/FilesOrchestrator.swift

```diff
@@ -157,13 +157,11 @@ internal class FilesOrchestrator: FilesOrchestratorType {
                 .compactMap { try deleteFileIfItsObsolete(file: $0.file, fileCreationDate: $0.creationDate) }
                 .sorted(by: { $0.creationDate < $1.creationDate })
 
-            #if DD_SDK_COMPILED_FOR_TESTING
```

Comment:
Are we sure about this? When we do this "open" batch is going to be included.

Reply:
This logic is conditioned by `ignoreFilesAgeWhenReading` flag enabled from [`core.flushAndTearDown()`](https://github.com/DataDog/dd-sdk-ios/blob/36862590661791e1ac6a0d61fbaf863c94cea97d/DatadogCore/Sources/Core/DatadogCore.swift#L206-L208) which is only available when `DD_SDK_COMPILED_FOR_TESTING` macro is set:
https://github.com/DataDog/dd-sdk-ios/blob/36862590661791e1ac6a0d61fbaf863c94cea97d/DatadogCore/Sources/Datadog.swift#L485-L489

so I don't see how this may interrupt `DataUploadWorker` in production apps. The entire concept of "ignoring files age" was introduced for testing purpose (we want to force-send all data in tests). Does it sound ok to you @maciejburda?

Ultimately, I don't like using macros in this way as they shred the codebase. Since it was introduced a long time ago however, we managed to keep the impact of `DD_SDK_COMPILED_FOR_TESTING` very minimal. A better approach would be to abstract certain components and inject different implementation in tests - it implies a much bigger effort, but I hope we do it one day.

Reply:
Not sure what am I missing, but the modified function `getReadableFiles(excludingFilesNamed excludedFileNames: Set<String> = [], limit: Int = .max)` is being called by the `DataUploadWorker``s `FileReader`.

Reply:
The logic we enable here is conditioned by `ignoreFilesAgeWhenReading` flag 🙂:
```swift
if ignoreFilesAgeWhenReading {
   return filesFromOldest
      .prefix(limit)
      .map { $0.file }
}
```
And this is only set to `true` in our tests 💡.

Reply:
Ah, I get it now! Thanks for clarifying :)